checkstyle/src/main.rs

```diff
@@ -0,0 +1,124 @@
+use regex::Regex;
+use std::fs::{self};
+
+fn get_file(filename: &str) -> String {
+    fs::read_to_string(filename).expect("Failed to read file")
+}
+
+fn split_into_lines(s: &str) -> Vec<&str> {
+    s.lines().collect()
+}
```

Comment:
You only ever iterate over the lines. Why do this dance converting iterator into collection into iterator? `for line in file.lines()` is sufficient and more direct.